src/components/checkbox/checkbox.js

```diff
@@ -156,11 +175,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
           listener(ev);
         }
       }
+
       function listener(ev) {
         if (element[0].hasAttribute('disabled')) {
           return;
         }
 
+        if (indeterminateStateEnabled) {
+          if (attr.indeterminateClick) {
+            scope.$apply(element.attr('indeterminate-click'));
```

Comment:
Yes, eval would be the correct thing to use here, and I should have noticed the `if` statement had the `attr` I needed :+1:.

As a sidenote: I tried substituting `$evalAsync` in here but it breaks necessary indet state logic.

This is because if this runs asynchronously it does not skip the code I need it to in the `render` function which ensures when a checkbox is indeterminate it does not adhere to the normal "checked' / 'unchecked' click logic.